From 79a426684b1b204ea766713fb0f10da9ffaa99be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20Wei=C3=9F?= <michael.weiss@aisec.fraunhofer.de>
Date: Fri, 14 Dec 2018 17:38:09 +0000
Subject: [PATCH] dont-depend-on-help2man

Upstream-Status: Inappropriate
---
 Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c8fe9e3..7c1039f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,8 +1,8 @@
-EXTRA_DIST = README openssl.cnf.sample create_tpm2_key.1
+EXTRA_DIST = README openssl.cnf.sample
 
-man1_MANS = create_tpm2_key.1
+#man1_MANS = create_tpm2_key.1
 
-CLEANFILES = $(man1_MANS)
+#CLEANFILES = $(man1_MANS)
 
 openssl_engine_LTLIBRARIES=libtpm2.la
 bin_PROGRAMS=create_tpm2_key
-- 
2.19.2

