From f7dd155aa0a70962c9c88a5e406e481599ea4b4f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20Wei=C3=9F?= <michael.weiss@aisec.fraunhofer.de>
Date: Fri, 10 Jan 2020 17:34:18 +0100
Subject: [PATCH] shiftfs: allow mounting of other shiftfs on shiftfs

---
 fs/shiftfs.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/shiftfs.c b/fs/shiftfs.c
index d0f29b4b7aee..8f1e312fd4b4 100644
--- a/fs/shiftfs.c
+++ b/fs/shiftfs.c
@@ -48,6 +48,8 @@ static void shiftfs_fill_inode(struct inode *inode, unsigned long ino,
 #define SHIFTFS_PASSTHROUGH_ALL                                                \
 	(SHIFTFS_PASSTHROUGH_STAT | SHIFTFS_PASSTHROUGH_IOCTL)
 
+#define SHIFTFS_MAX_STACK_DEPTH 3
+
 static inline bool shiftfs_passthrough_ioctls(struct shiftfs_super_info *info)
 {
 	if (!(info->passthrough & SHIFTFS_PASSTHROUGH_IOCTL))
@@ -1928,7 +1930,7 @@ static int shiftfs_fill_super(struct super_block *sb, void *raw_data,
 	}
 
 	sb->s_stack_depth = dentry->d_sb->s_stack_depth + 1;
-	if (sb->s_stack_depth > FILESYSTEM_MAX_STACK_DEPTH) {
+	if (sb->s_stack_depth > SHIFTFS_MAX_STACK_DEPTH) {
 		printk(KERN_ERR "shiftfs: maximum stacking depth exceeded\n");
 		err = -EINVAL;
 		goto out_put_path;
-- 
2.11.0

