From 1505992c329e310d748cba466543dbbada3b35ea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20Wei=C3=9F?= <michael.weiss@aisec.fraunhofer.de>
Date: Tue, 21 May 2019 11:20:06 +0000
Subject: [PATCH] Makefile: remove -march=armv7-a from ARCH arm

---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index ccda445..8a42797 100644
--- a/Makefile
+++ b/Makefile
@@ -163,7 +163,6 @@ ifeq ($(ARCH), arm)
 	OBJS		+= arm/aarch32/kvm-cpu.o
 	ARCH_INCLUDE	:= $(HDRS_ARM_COMMON)
 	ARCH_INCLUDE	+= -Iarm/aarch32/include
-	CFLAGS		+= -march=armv7-a
 
 	ARCH_WANT_LIBFDT := y
 endif
-- 
2.11.0

