From e64160987461c3f67b9b73805c944c494fa0aa00 Mon Sep 17 00:00:00 2001
From: Andrea Righi <andrea.righi@canonical.com>
Date: Wed, 27 Apr 2022 18:20:41 +0200
Subject: [PATCH 32/35] UBUNTU: SAUCE: shiftfs: fix missing include required in
 5.18

Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 fs/shiftfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/shiftfs.c b/fs/shiftfs.c
index a21624c529f0..a5338dc6290c 100644
--- a/fs/shiftfs.c
+++ b/fs/shiftfs.c
@@ -21,6 +21,7 @@
 #include <linux/posix_acl_xattr.h>
 #include <linux/uio.h>
 #include <linux/fiemap.h>
+#include <linux/pagemap.h>
 
 struct shiftfs_super_info {
 	struct vfsmount *mnt;
-- 
2.30.2

