From 58c920d84118f62fd98f39ac526ed42389138b30 Mon Sep 17 00:00:00 2001
From: Andrea Righi <andrea.righi@canonical.com>
Date: Tue, 18 Oct 2022 17:09:12 +0200
Subject: [PATCH 33/35] UBUNTU: SAUCE: shiftfs: support kernel 6.1

Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 fs/shiftfs.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/fs/shiftfs.c b/fs/shiftfs.c
index a5338dc6290c..34f080ae0fec 100644
--- a/fs/shiftfs.c
+++ b/fs/shiftfs.c
@@ -747,19 +747,18 @@ static int shiftfs_fiemap(struct inode *inode,
 }
 
 static int shiftfs_tmpfile(struct user_namespace *ns,
-			   struct inode *dir, struct dentry *dentry,
+			   struct inode *dir, struct file *file,
 			   umode_t mode)
 {
 	int err;
 	const struct cred *oldcred;
-	struct dentry *lowerd = dentry->d_fsdata;
 	struct inode *loweri = dir->i_private;
 
 	if (!loweri->i_op->tmpfile)
 		return -EOPNOTSUPP;
 
 	oldcred = shiftfs_override_creds(dir->i_sb);
-	err = loweri->i_op->tmpfile(&init_user_ns, loweri, lowerd, mode);
+	err = loweri->i_op->tmpfile(&init_user_ns, loweri, file, mode);
 	revert_creds(oldcred);
 
 	return err;
-- 
2.30.2

