From 36f70faba13416f21b274fcae1631b317fe9878e Mon Sep 17 00:00:00 2001
From: Andrea Righi <andrea.righi@canonical.com>
Date: Wed, 4 Jan 2023 10:25:30 +0100
Subject: [PATCH 34/35] UBUNTU: SAUCE: shiftfs: support linux 6.2

Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 fs/shiftfs.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/fs/shiftfs.c b/fs/shiftfs.c
index 34f080ae0fec..cda74b614505 100644
--- a/fs/shiftfs.c
+++ b/fs/shiftfs.c
@@ -912,7 +912,7 @@ shiftfs_get_acl(struct inode *inode, int type, bool rcu)
 		return NULL;
 
 	oldcred = shiftfs_override_creds(inode->i_sb);
-	lower_acl = get_acl(loweri, type);
+	lower_acl = get_inode_acl(loweri, type);
 	revert_creds(oldcred);
 
 	if (lower_acl && !IS_ERR(lower_acl)) {
@@ -1026,13 +1026,13 @@ static const struct inode_operations shiftfs_dir_inode_operations = {
 	.permission	= shiftfs_permission,
 	.getattr	= shiftfs_getattr,
 	.listxattr	= shiftfs_listxattr,
-	.get_acl	= shiftfs_get_acl,
+	.get_inode_acl	= shiftfs_get_acl,
 };
 
 static const struct inode_operations shiftfs_file_inode_operations = {
 	.fiemap		= shiftfs_fiemap,
 	.getattr	= shiftfs_getattr,
-	.get_acl	= shiftfs_get_acl,
+	.get_inode_acl	= shiftfs_get_acl,
 	.listxattr	= shiftfs_listxattr,
 	.permission	= shiftfs_permission,
 	.setattr	= shiftfs_setattr,
@@ -1041,7 +1041,7 @@ static const struct inode_operations shiftfs_file_inode_operations = {
 
 static const struct inode_operations shiftfs_special_inode_operations = {
 	.getattr	= shiftfs_getattr,
-	.get_acl	= shiftfs_get_acl,
+	.get_inode_acl	= shiftfs_get_acl,
 	.listxattr	= shiftfs_listxattr,
 	.permission	= shiftfs_permission,
 	.setattr	= shiftfs_setattr,
-- 
2.30.2

